src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs

```diff
@@ -56,11 +56,52 @@ public static ICollection<ISymbol> GetMissingEnumMembers(ISwitchExpressionOperat
 
                     enumMembers.Remove(IntegerUtilities.ToInt64(constantValue.Value));
                 }
+                else if (arm.Pattern is IBinaryPatternOperation binaryPattern)
+                {
+                    HandleBinaryPattern(binaryPattern, enumMembers);
+                }
             }
 
             return true;
         }
 
+        private static void HandleBinaryPattern(IBinaryPatternOperation binaryPattern, Dictionary<long, ISymbol> enumMembers)
+        {
+            if (binaryPattern.OperatorKind == BinaryOperatorKind.Or)
+            {
+                if (!RemoveIfConstantPatternHasValue(binaryPattern.LeftPattern))
+                {
+                    if (binaryPattern.LeftPattern is IBinaryPatternOperation leftBinaryPattern)
+                    {
+                        HandleBinaryPattern(leftBinaryPattern, enumMembers);
+                    }
+                }
+
+                if (!RemoveIfConstantPatternHasValue(binaryPattern.RightPattern))
+                {
+                    if (binaryPattern.RightPattern is IBinaryPatternOperation rightBinaryPattern)
+                    {
+                        HandleBinaryPattern(rightBinaryPattern, enumMembers);
+                    }
+                }
+            }
+
+            bool RemoveIfConstantPatternHasValue(IOperation operation)
+            {
+                if (operation is IConstantPatternOperation constantPattern)
+                {
+                    var constantValue = constantPattern.Value.ConstantValue;
+                    if (constantValue.HasValue)
+                    {
+                        enumMembers.Remove(IntegerUtilities.ToInt64(constantValue.Value));
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
```

Comment:
```suggestion
        private static void HandleBinaryPattern(IBinaryPatternOperation? binaryPattern, Dictionary<long, ISymbol> enumMembers)
        {
            if (binaryPattern?.OperatorKind == BinaryOperatorKind.Or)
            {
                RemoveIfConstantPatternHasValue(binaryPattern.LeftPattern);
                RemoveIfConstantPatternHasValue(binaryPattern.RightPattern);
                
                HandleBinaryPattern(binaryPattern.LeftPattern as IBinaryPatternOperation);
                HandleBinaryPattern(binaryPattern.RightPattern as IBinaryPatternOperation);
            }

            void RemoveIfConstantPatternHasValue(IOperation operation)
            {
                if (operation is IConstantPatternOperation { Value { ConstantValue { HasValue: true, Value: var value } } })
                        enumMembers.Remove(IntegerUtilities.ToInt64(value));
            }
        }
```

Reply:
@CyrusNajmabadi That's ALOT better! Loved how the code transformed into a much readable and neater code.